File: q7/q7.py
class Solution:
    def reverse(self, x: int) -> int:
        solution = 0
        if(x > 0):
            positive = 1
        else:
            positive = 0
            x = 0 - x
        while x != 0:
            pop = x % 10
            x = int(x/10)
            solution = solution *10 + pop
        if positive and solution <= 2**31 - 1:
            return solution
        elif positive < 1 and solution < 2**31 - 1:
            return (-solution) 
        return 0

File: Sequence/q1-q99/q1-q9/q7/q7.cpp
class Solution {
public:
    int reverse(int x) {
        int solution = 0;
        while(x != 0){
            int pop = x % 10;
            x /= 10;
            if (solution > INT_MAX/10 || (solution == INT_MAX / 10 && pop > 7)) return 0;
            if (solution < INT_MIN/10 || (solution == INT_MIN / 10 && pop < -8)) return 0;
            solution = solution * 10 + pop;
        }
        return solution;
    }
};
